sim/paladin/soc.go

```diff
@@ -0,0 +1,157 @@
+const socProcSpellCoeff = 0.29
+const socJudgeSpellCoeff = 0.423
+
+var socLevels = [socRanks + 1]int{0, 20, 30, 40, 50, 60}
+var socManaCosts = [socRanks + 1]float64{0, 65, 110, 140, 180, 210}
+var socAuraSpellIDs = [socRanks + 1]int32{0, 20375, 20915, 20918, 20919, 20920}
+var socProcSpellIDs = [socRanks + 1]int32{0, 20424, 20944, 20945, 20946, 20947}
+var socJudgeSpellIDs = [socRanks + 1]int32{0, 20467, 20963, 20964, 20965, 20966}
+var socJudgeBasePoints = [socRanks + 1]float64{0, 92, 145, 203, 260, 338}
+var socJudgeRealPointsPerLevel = [socRanks + 1]float64{0, 5.6, 6.1, 5.6, 6.1, 6.1}
+var socEffectDieSides = [socRanks + 1]float64{0, 9, 15, 21, 27, 35}
+var socLevelMinMaxEffects = [socRanks + 1][]int32{{0}, {20, 28}, {30, 38}, {40, 48}, {50, 58}, {60, 60}}
+
+func (paladin *Paladin) applySealOfCommandSpellAndAuraBaseConfig(rank int) {
+	spellIDProc := socProcSpellIDs[rank]
+	spellIDAura := socAuraSpellIDs[rank]
+	spellIDJudge := socJudgeSpellIDs[rank]
+	manaCost := socManaCosts[rank]
+	level := socLevels[rank]
+	scalingLevelMin := socLevelMinMaxEffects[rank][0]
+	scalingLevelMax := socLevelMinMaxEffects[rank][1]
+	judgeBasePoints := socJudgeBasePoints[rank]
+	judgePointsPerLevel := socJudgeRealPointsPerLevel[rank]
+	judgeDieSides := socEffectDieSides[rank]
+
+	levelsToScale := min(paladin.Level, scalingLevelMax) - scalingLevelMin
+	judgeMinDamage := judgeBasePoints + float64(levelsToScale)*judgePointsPerLevel
+	judgeMaxDamage := judgeMinDamage + judgeDieSides
+
+	onJudgementProc := paladin.RegisterSpell(core.SpellConfig{
+		ActionID:    core.ActionID{SpellID: spellIDJudge}, // Judgement of Righteousness.
+		SpellSchool: core.SpellSchoolHoly,
+		ProcMask:    core.ProcMaskMeleeSpecial,
+		Flags:       core.SpellFlagMeleeMetrics | SpellFlagSecondaryJudgement,
+
+		DamageMultiplier: 1,
+		ThreatMultiplier: 1,
+		CritMultiplier:   paladin.SpellCritMultiplier(),
+		BonusCritRating:  paladin.getBonusCritChanceFromHolyPower(),
+
+		ApplyEffects: func(sim *core.Simulation, target *core.Unit, spell *core.Spell) {
+			baseDamage := sim.Roll(judgeMinDamage, judgeMaxDamage) + socJudgeSpellCoeff*spell.SpellDamage()
+			spell.CalcAndDealDamage(sim, target, baseDamage, spell.OutcomeMeleeSpecialCritOnly)
+		},
+	})
+
+	onSwingProc := paladin.RegisterSpell(core.SpellConfig{
+		ActionID:         core.ActionID{SpellID: spellIDProc},
+		SpellSchool:      core.SpellSchoolHoly,
+		ProcMask:         core.ProcMaskMeleeMHSpecial | core.ProcMaskSuppressedExtraAttackAura, // This needs figured out properly
+		Flags:            core.SpellFlagMeleeMetrics,
+		RequiredLevel:    level,
+		DamageMultiplier: 0.7,
+		ThreatMultiplier: 1.0,
+		CritMultiplier:   paladin.MeleeCritMultiplier(),
+
+		ApplyEffects: func(sim *core.Simulation, target *core.Unit, spell *core.Spell) {
+			baseDamage := -1 + spell.Unit.MHWeaponDamage(sim, spell.MeleeAttackPower())
+			fullDamage := baseDamage + socProcSpellCoeff*spell.SpellPower() + spell.BonusWeaponDamage()
```

Comment:
SpellPower instead of SpellDamage seems to be wrong

Reply:
Fixed in https://github.com/wowsims/sod/pull/302/commits/553a78db7093e4aeba162e7b6a5ed6686bc1cf68